### PR TITLE
Document how app developers can use docker compose

### PR DESCRIPTION
**Pull request by mrjones-plip**
This PR:

* Adds a new page documenting how to use the new `docker-compose-developer.yml` file
* Makes minor updates/corrections to `core/guides/docker-setup` page
* Adds an intro section on the `apps/guides/hosting` page

per #medic/cht-core/issues/7211

### CODE REVIEW

**Review by garethbowen — COMMENTED** [review]
Great work @mrjones-plip ! I haven't done a full review, but I think it would be good to be more clear that the app-developer version is not for use in production. Either add a bit to the note at the top, or add a paragraph to "getting started" defining who should use this file and why it exists?

**Comment by mrjones-plip** [comment]
Cool cool - thanks for early feedback!  I'll be sure to call this out.

**Comment by mrjones-plip** [comment]
@garethbowen  - added 7e04a69 to tend to this in the note.

File: content/en/apps/guides/hosting/app-developer.md
Comment (newtewt) [inline content/en/apps/guides/hosting/app-developer.md:30]: Might explain what docker is doing that it might be taking long to pull the images. 

`This may take a while as docker is pulling the images and starting the application. This is based on the speed of the Internet connection and speed of the bare-metal host`

File: content/en/apps/guides/hosting/app-developer.md
Comment (newtewt) [inline content/en/apps/guides/hosting/app-developer.md:71]: Need either to reference the `yml` file in the directory above or add a `cp` command to copy it through to the new directory.  I don't think docker will traverse the directories right?

File: content/en/apps/guides/hosting/app-developer.md
Comment (newtewt) [inline content/en/apps/guides/hosting/app-developer.md:78]: Any reason to not clue them in on the `-d` argument? Runs detached. 

`docker-compose --env-file ../the_second/.env-docker-compose -f docker-compose-developer.yml up -d`

File: content/en/apps/guides/hosting/app-developer.md
Comment (newtewt) [inline content/en/apps/guides/hosting/app-developer.md:52]: `A nice compliment to this is to store the specifics on how to run the`

File: content/en/apps/guides/hosting/app-developer.md
Comment (newtewt) [inline content/en/apps/guides/hosting/app-developer.md:95]: Unless it has changed recently. The incognito modes didn't play well with our app due to local storage access.

**Review by newtewt — CHANGES_REQUESTED** [review]
Some questions and minor changes inline. This is cool.

File: content/en/apps/guides/hosting/app-developer.md
Comment (mrjones-plip) [reply content/en/apps/guides/hosting/app-developer.md:71]: Turns out it will! This is literal code I ran to ensure it work.

File: content/en/apps/guides/hosting/app-developer.md
Comment (mrjones-plip) [reply content/en/apps/guides/hosting/app-developer.md]: Cool cool! I added a one liner in there per 1544ed0

File: content/en/apps/guides/hosting/app-developer.md
Comment (mrjones-plip) [reply content/en/apps/guides/hosting/app-developer.md:78]: Hmmm - good question! Checking now I see in all our docs we don't mention the `-d` option except on the [Self Hosted](https://docs.communityhealthtoolkit.org/apps/guides/hosting/self-hosting/) section (not even on our [helpful docker commands](https://docs.communityhealthtoolkit.org/core/guides/docker-setup/#helpful-docker-commands)!). I think this is fine for now until we do a holistic audit on how to improve. 

Feel free to push back!

File: content/en/apps/guides/hosting/app-developer.md
Comment (mrjones-plip) [reply content/en/apps/guides/hosting/app-developer.md:52]: I'll wait to hear back on how this sentence ends ;) 

otherwise resolving for now to tidy it up so I can send it back to you.

File: content/en/apps/guides/hosting/app-developer.md
Comment (mrjones-plip) [reply content/en/apps/guides/hosting/app-developer.md]: D'oh!  Nice catch - thanks for letting me know.  fixed in 524bfd2

File: content/en/apps/guides/hosting/app-developer.md
Comment (newtewt) [reply content/en/apps/guides/hosting/app-developer.md:52]: Sorry, there was an added to.  It reads a little weird otherwise. 

` A nice compliment to this is store the specifics on how to run the`
 
 `A nice compliment to this is to store the specifics on how to run the`

**Review by newtewt — APPROVED** [review]
I updated one of my comments. I should have mentioned I added a word to the sentence so it wasn't confusing.  Otherwise it looks good to go.  I'm approving now to not hold anything up.

File: content/en/apps/guides/hosting/app-developer.md
Comment (mrjones-plip) [reply content/en/apps/guides/hosting/app-developer.md]: D'oh - That was on me - I totally didn't follow that this was a comment to fix my own content - I thought you were telling me a "nice compliment to this..." :facepalm: Sorry!

Fixed in  c100145 !

**Comment by mrjones-plip** [comment]
blocked on [this PR](https://github.com/medic/cht-core/pull/7212) being merged in `cht-core` before this docs PR can be merged and go live.

**Pull request merged**
